### PR TITLE
Deliver all completed outputs

### PR DESCRIPTION
**Pull request by tonymontaro**
#### What does this PR do?
This PR contains output for the following:
- checking repository health
- setting up continuous integration with Travis CI
- setting up software development environments
- getting peer reviews
- setting up the repository
- reflecting on the learning experience
- delivering your product
- using design patterns
- writing front end template (html, css, js)

#### Description of Tasks to be completed?
Provide:
- a screenshot of your code coverage and GPA badges in your repo's README file,
- a link to a successful TravisCI build of your development branch (it should include test runs),
- live URLs to a review app, your staging & production apps
- a Pull Request showing at least two Peer Code Reviews
- the repository for your proposed project and at least one Pull Request
- a post reflecting on your learning experience while producing these outputs
- Release notes for your final product in your repo's wiki
- a detailed explanation of Design Patterns used while solving-the-problem
- a folder containing the template for the proposed frontend of your project

#### How should this be manually tested?
For:
- checking repository health: visit the [output folder](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-angene/06-building-a-program-start-to-finish/checking-repository-health/output), it contains the screenshot and link 
- setting up continuous integration with Travis CI: open the successful Travis [build link](https://travis-ci.org/andela-angene/inverted-index)
- setting up software development environments: open the [review](https://tony-invertedindex-stagin-pr-7.herokuapp.com/), [staging](http://tony-invertedindex-staging.herokuapp.com/) and [production](http://tony-invertedindex-production.herokuapp.com/) links
- getting peer reviews: open the [link](https://github.com/andela-angene/inverted-index/pull/5) to a pull request with peer reviews
- setting up the repository: open the [repository link](https://github.com/andela-angene/inverted-index)
- reflecting on the learning experience: read the [blog post](https://medium.com/@ngeneanthony/the-simulations-c021fd035a10) on the subject
- delivering your product: open the [release notes](https://docs.google.com/document/d/1EgBswRnfhx9XSfaXmHTKMYDZykrQK2hjyxD4iJw2yGk/)
- using design patterns: visit the [output folder](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-angene/06-building-a-program-start-to-finish/using-design-patterns/output), it contains the explanation
- writing front end template (html, css, js): visit the [output folder](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-angene/06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output), it contains the front-end template

### CODE REVIEW

File: 06-building-a-program-start-to-finish/checking-repository-health/output/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#### Output
+- **Using**: `Code Climate` and `Coveralls`
+- **Deliver**: a screenshot of your code coverage and GPA badges in your repo's README file
+- **With**: *independently*
+
+#### Screenshot
+- ![code-climate-coverage](http://i.imgur.com/pNJT71h.png)
+
+#### Link
+- [Repo](https://github.com/andela-angene/inverted-index)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/checking-repository-health/output/README.md:10]: Looks good!

File: 06-building-a-program-start-to-finish/delivering-your-product/output/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: Google Doc
+- **Deliver**: `Release notes` for your final product in your repo's wiki
+- **With**: *independently*
+
+#### Link
+- [Release Notes](https://docs.google.com/document/d/1EgBswRnfhx9XSfaXmHTKMYDZykrQK2hjyxD4iJw2yGk/)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/delivering-your-product/output/README.md:7]: Looks good!

File: 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: GitHub
+- **Deliver**: a Pull Request showing at least two Peer Code Reviews
+- **With**: *independently*
+
+####  Pull Request with Peer Reviews
+- [Link](https://github.com/andela-angene/inverted-index/pull/5)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md:7]: Cool!

File: 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/output/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: Medium
+- **Deliver**: a post reflecting on your learning experience while producing these outputs
+- **With**: *independently*
+
+#### Link
+- [Medium Post](https://medium.com/@ngeneanthony/the-simulations-c021fd035a10)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/output/README.md:7]: On to the next phase we look! :simple_smile:

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: TravisCI
+- **Deliver**: a link to a successful TravisCI build of your `development` branch (it should include test runs)
+- **With**: *independently*
+
+#### Link
+- Successful [Travis Build](https://travis-ci.org/andela-angene/inverted-index)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md:7]: All green!

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#### Output
+- **Using**: Heroku
+- **Deliver**: live URLs to a `review` app, your `staging` & `production` apps
+- **With**: *independently*
+
+#### Links
+- [Review](http://tony-invertedindex-review.herokuapp.com/)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md]: I don't believe this to be a review app. It's supposed to be named according to PRs raised in your repo. Please review this.

File: 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: GitHub
+- **Deliver**: the repository for your proposed project and at least one `Pull Request`
+- **With**: *independently*
+
+#### Link
+- [Repository](https://github.com/andela-angene/inverted-index)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md:7]: Please ensure to clearly outline the features of the application in the ReadMe and how to use it.

File: 06-building-a-program-start-to-finish/using-design-patterns/output/README.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+}
+```
+An we can create a new instance of `InvertedIndex` by calling the constructor thus:
+```javascript
+const index = new InvertedIndex();
+
+index.validateFile(fileContent);
+
+index.createIndex(fileContent, 'books.json');
+```
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/using-design-patterns/output/README.md:48]: Looks good!

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/README.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
+#### Output
+- **Using**: HTML, CSS & JS
+- **Deliver**: a folder containing the template for the proposed frontend of your project
+- **With**: *independently*
+
+#### Screenshots
+- ![InvertedIndex1](http://i.imgur.com/5RbodoH.png)
+- ![InvertedIndex2](http://i.imgur.com/MIy5P7m.png)
+- ![InvertedIndex3](http://i.imgur.com/iqFLtBD.png)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/README.md:9]: Seen!

**Review by q-ode — COMMENTED** [review]
Great Job!

Please attend to the comments.

File: 06-building-a-program-start-to-finish/checking-repository-health/output/README.md (hunk 1 above)
Comment (tonymontaro) [reply 06-building-a-program-start-to-finish/checking-repository-health/output/README.md:10]: Cool.

File: 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/output/README.md (hunk 4 above)
Comment (tonymontaro) [reply 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/output/README.md:7]: Lol, onto the next one.

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md (hunk 5 above)
Comment (tonymontaro) [reply 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md:7]: Awesome!

**Comment by tonymontaro** [comment]
I made the changes, thanks.

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md (hunk 6 above)
Comment (tonymontaro) [reply 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md]: I added the correct review app link generated by Heroku pipeline.

File: 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md (hunk 7 above)
Comment (tonymontaro) [reply 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md:7]: The README file has been updated.

File: 06-building-a-program-start-to-finish/using-design-patterns/output/README.md (hunk 8 above)
Comment (tonymontaro) [reply 06-building-a-program-start-to-finish/using-design-patterns/output/README.md:48]: Thanks.

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/README.md (hunk 9 above)
Comment (tonymontaro) [reply 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/README.md:9]: Thanks.

**Pull request merged**